src/DataTrail.tsx

```diff
@@ -171,6 +183,14 @@ export class DataTrail extends SceneObjectBase<DataTrailState> implements SceneO
   private _addingFilterWithoutReportingInteraction = false;
   private datasourceHelper = new MetricDatasourceHelper(this);
 
+  private _isAddToDashboardServiceAvailable(): boolean {
+    try {
+      return typeof getAddToDashboardService === 'function' && getAddToDashboardService() != null;
+    } catch {
+      return false;
+    }
```

Comment:
If `getAddToDashboardService` never throws, we could just use:

```suggestion
    return typeof getAddToDashboardService === 'function' && getAddToDashboardService() != null;
```